<center><h1>Звіт Лабораторна робота №2</h1></center>

<center><h2>Виконав студент групи РА-241 Жук Дмитро</h2></center>

<h4>Мета роботи - дослідження амплітудних та фазових співвідношень у ланцюгах змінного струму, частотних характеристик та резонансних явищ, побудова векторних діаграм струмів та напруг.</h4>

При виконанні роботи вивчатимемо схему, наведену на малюнку 2.1

<center><img src="images/WINWORD_Gcuhxfo8Yi.png" width="450"/> <h4>Рисунок 2.1 - Електрична схема</h4></center>


Двополюсники, які мають бути досліджені в роботі, показані на (рис. 2.2).

<center><img src="images/WINWORD_OT27yfa5sk.png" width="450"/> <h4>Рисунок 2.2</h4></center>

При дослідженнях ці ланцюги включають між точками А та В згідно зі схемою рис. 2.1. Значення індуктивностей La, Lb = 10 мГн та ємностей Ca, Cb = 33 нФ, опір R1 = 100 Ом.

<center><h4>Хід роботи</h4><center></center>

У програмному середовищі TINA зберемо схему, що вивчається (рис. 2.1). 


У генераторі напруги VG1 необхідно встановити синусоїдальну форму сигналу, частотою 5 кГц, амплітудою 1, фазою 0°.

Далі налаштуємо осцилограф. На вкладці головного меню Т&М виберемо пункт меню "Осцилограф" ("Oscilloscope"). 
Встановимо горизонтальну розгортку 50 мкс, VM1 – вертикальну чутливість 1 В/поділ., VM2 – вертикальну чутливість 1 В/поділ., режим синхронізації встановимо "Single". Для того, щоб відобразити на екрані осцилографа сигнали зі схеми, треба у вікні "Channel" вибрати сигнал, у вікні "Vertical" встановити потрібну чутливість і у вікні "Coupling" встановити DC або AC (залежно від форми сигналу) і натиснути ON. Якщо для вибраного сигналу встановити невірну чутливість, сигнал не відображатиметься на осцилографі. Щоб запустити відображення на екрані осцилографа, необхідно натиснути кнопку "Run". 

<center><img src="images/chrome_BuxE3mbWNJ.png" width="800"/> <h4>Рисунок 2.3 - Графіки напруги</h4></center>

Підключимо між точками А і B індуктивність La і розрахуємо частоту сигналу генератора f1, коли реактивний опір індуктивності XL дорівнює опору R1. Для цього використаємо бібліотеку SumPy.

$$ X_L = 2 \cdot \pi \cdot f_1 \cdot L_a = R_1 $$

In [2]:
import sympy as smp      # Імпортуємо бібліотеку SymPy для символьних обчислень
import math              # Імпортуємо math, щоб мати доступ до константи π

# Задаємо відомі числові значення:
R1_s = 100              
La_s = 0.01              

# Оголошуємо символьні змінні для формули
X_L, f1, R1, La = smp.symbols('X_L f1 R1 La')

# Оскільки X_L = R1, то 2πf1La = R1
eq1 = smp.Eq(2 * math.pi * f1 * La, R1)

# Розв’язуємо рівняння відносно змінної f1 (частоти)
eq1_solve = smp.solve(eq1, f1)

# Підставляємо числові значення R1 = 100, La = 0.01 у знайдений розв’язок
eq1_sub = eq1_solve[0].subs(R1, R1_s).subs(La, La_s)

# Виводимо отримане значення частоти f1
print(F"Частота сигналу генератора f1, коли X_L = R1: {eq1_sub:.1f} Гц")

Частота сигналу генератора f1, коли X_L = R1: 1591.5 Гц


Встановимо напругу генератора VG1, що дорівнює 1 В, а частоту дорівнює 0,1f1. Значення DX = ХB - ХA = Т2 - Т1 визначає часовий зсув осцилограми.

<center><img src="images/chrome_WTFKGhDbAd.png" width="800"/> <h4>Рисунок 2.4</h4></center>

Розрахуємо різницю фаз за такою формулою:
$$
\varphi = 2 \cdot \pi \cdot f \cdot (T2 - T1)
$$

In [51]:
# Задаємо відомі числові значення:
f = 1591.15              
DX = 0.000100             

# Розв’язуємо рівняння
fi = 2 * math.pi * f * DX

# Виводимо отримане значення 
print(F"Різниця фаз: {fi:.4f} рад")

Різниця фаз: 0.9997 рад


Вибравши пункти меню "Аналіз - Аналіз змінного струму - Таблиця результатів змінного струму", запишемо значення для $U_{VМ1}$, $U_{VМ2}$ = 1,22 В та $I_{L1}$ = 9,95 мА.

Через пункти меню ("Analysis - Time function") встановивши (Start time = 0), а "Закінчити показ" (End time) рівним приблизно 2 - 3 періодів досліджуваної частоти отримаємо графіки напруг VМ1 і VМ2. У нашому випадку період T дорівнюватиме $T = 1 / 0,1 \cdot f_1$. отримане значення необхідно збільшити 2 – 3 рази (рис. 2.5).

$T_1 = \dfrac{1}{0,1 \cdot f_1} \cdot 3 = \dfrac{1}{0,1 \cdot 1591.5} \cdot 3 = 18.85  $ $мілісекунд.$

<center><img src="images/TINA_tB1cAddDpN.png" width="400"/> <h4>Рисунок 2.5 - Графіки напруг VМ1 і VМ2 ($0,1 \cdot f_1$)</h4></center>

Повторимо дослідження для частот $f_1$ та $10 \cdot f_1$.

$T_2 = \dfrac{1}{f_1} \cdot 3 = \dfrac{1}{1591.5} \cdot 3 = 1,885  $ $мілісекунд.$

<center><img src="images/TINA_6F0MLZOKxg.png" width="400"/> <h4>Рисунок 2.6 - Графіки напруг VМ1 і VМ2 ($f_1$)</h4></center>

$T_3 = \dfrac{1}{ 10 \cdot f_1} \cdot 3 = \dfrac{1}{ 10 \cdot 1591.5} \cdot 3 = 0,1885  $ $мілісекунд.$

<center><img src="images/TINA_bBqdEaG48w.png" width="400"/> <h4>Рисунок 2.7 - Графіки напруг VМ1 і VМ2 ($10 \cdot f_1$)</h4></center>

Вибравши в головному меню ("Analysis - AC Analysis - AC Transfer characteristic"), встановлюємо діапазон частот від $0,1 \cdot f_1$ до $10 \cdot f_1$, з лінійною розгорткою тип діаграми - "Амплітуда та Фаза" і отримуємо графіки АЧХ та ФЧХ (рис. 2.8).

<center><img src="images/TINA_MEYZpewO0m.png" width="400"/> <h4>Рисунок 2.8 - Графіки АЧХ та ФЧХ $L_a$ (Лінійна розгортка)</h4></center>

Також побудуємо графіки АЧХ та ФЧХ для логарифмічної розгортки (рис. 2.9).

<center><img src="images/TINA_a8orHAQRn2.png" width="400"/> <h4>Рисунок 2.8 - Графіки АЧХ та ФЧХ $L_a$ (Логарифмічна розгортка)</h4></center>

Увімкнемо між точками А та B ємність $С_а$ та розрахуємо частоту сигналу генератора $f_2$, при якій реактивний опір ємності дорівнює опору R1 (R1 = 100 Ом). Для цього як і раніше використаємо бібліотеку SumPy.

$$ X_C = \frac{1}{2 \cdot \pi \cdot f_2 \cdot C_a} = R1 $$

In [5]:
# Задаємо відомі числові значення:
R1_s = 100              
Ca_s = 3.3 * 10 **-8              

# Оголошуємо символьні змінні для формули
X_C, f2, R1, Ca = smp.symbols('X_C f2 R1 Ca')

# Оскільки X_L = R1, то 2πf1La = R1
eq1 = smp.Eq(1 / (2 * math.pi * f2 * Ca), R1)

# Розв’язуємо рівняння відносно змінної f2 (частоти)
eq1_solve = smp.solve(eq1, f2)

# Підставляємо числові значення 
eq1_sub = eq1_solve[0].subs(R1, R1_s).subs(Ca, Ca_s)

# Виводимо отримане значення частоти f2
print(F"Частота сигналу генератора f1, коли X_C = R1: {eq1_sub:.1f} Гц")

Частота сигналу генератора f1, коли X_C = R1: 48228.8


**Повторимо дослідження виконані раніше, але для ємності $C_a$.**

In [6]:
eq1_solve

[0.159154943091895/(Ca*R1)]

Визначимо часовий зсув осцилограми (рис. 2.9).

<center><img src="images/TINA_vq2JVpHdQd.png" width="700"/> <h4>Рисунок 2.9 - Часовий зсув осцилограми</h4></center>

Запишемо значення для $U_{VМ1}$ = 1,22 В, $U_{VМ2}$ = 0,7036 В та $I_{C1}$ = 0,99502 мА.

$T_1 = \dfrac{1}{ 0,1 \cdot f_1} \cdot 3 = \dfrac{1}{ 0,1 \cdot 48228.8} \cdot 3 = 0,62203  $ $мілісекунд.$

<center><img src="images/TINA_lJiJIoKmNA.png" width="400"/> <h4>Рисунок 2.10 - Графіки напруг VМ1 і VМ2 ($0,1 \cdot f_1$)</h4></center>

$T_2 = \dfrac{1}{f_1} \cdot 3 = \dfrac{1}{48228.8} \cdot 3 = 62,203  $ $мікросекунд.$

<center><img src="images/TINA_tDxAuconi5.png" width="400"/> <h4>Рисунок 2.11 - Графіки напруг VМ1 і VМ2 ($f_1$)</h4></center>

$T_3 = \dfrac{1}{10 \cdot f_1} \cdot 3 = \dfrac{1}{10 \cdot 48228.8} \cdot 3 = 6,2203  $ $мікросекунд.$

<center><img src="images/TINA_dK57EtyApI.png" width="400"/> <h4>Рисунок 2.12 - Графіки напруг VМ1 і VМ2 ($10 \cdot f_1$)</h4></center>

Побудуємо графіки АЧХ та ФЧХ для ємності ($C_a$).

<center><img src="images/TINA_dms1R8ht3z.png" width="400"/> <img src="images/TINA_T09qHqdu2T.png" width="400"/> <h4>Рисунок 2.13 - Графіки АЧХ та ФЧХ $C_a$ (Лінійна та Логарифмічна розгортки)</h4></center>

Для дослідження RLC ланцюга увімкнемо між точками А та B послідовно індуктивність $L_a$ та ємність $C_a$ та розрахуємо частоту резонансу $f_3$ даного RLC ланцюга. 

$$ f_3 = \frac{1}{2 \cdot \pi \cdot \sqrt{L_a \cdot C_a}} $$

In [9]:
# Задаємо відомі числові значення:
L_a = 0.01              
C_a = 3.3 * 10 **-8             

# Розв’язуємо рівняння
f3 = 1 / (2 * math.pi * math.sqrt(L_a * C_a))

# Виводимо отримане значення 
print(F" Частота резонансу: {f3:.1f} Гц")

 Частота резонансу: 8761.2 Гц


Додамо до нашої схеми ще два вольтметри, як показано на (рис 2.14).

<center><img src="images/WINWORD_3KmNzDhWtk.png" width="400"/> <h4>Рисунок 2.14</h4></center>

Встановимо напругу генератора VG1 = 1 В, а частоту = $f_3$. Вибравши пункти меню ("Analysis - AC Analysis - Table of AC results"), запишемо значення напруги та фази для $U_{VM1}$, $U_{VM2}$, $U_{VM3}$, $U_{VM4}$ та $I_{L1}$ до таблиці 2.1. Також повторимо дослідження для частот $0,1 \cdot f_3$ і $10 \cdot f_3$.

Таблиця вимірів та розрахунків 2.1
| Частота, Гц | $U_{VM1}$, В | $\varphi_{VM1}$ | $U_{VM2}$, В | $\varphi_{VM2}$ | $U_{VM3}$, В | $\varphi_{VM3}$ | $U_{VM4}$, В | $\varphi_{VM4}$ | $I_{L1}, мА$ | $\varphi_{IL1}$ |
| :-- | :-- | :-- | :-- | :-- | :-- | :-- | :-- | :-- | :-- | :-- |
| 876.12 | 1.22 | -90° | 0.013 | -1.05° |  0.0714 | 88.95° |  1.23 | -91.05° | 0.183 | -1.05° |
| 8761.2 | 1.22 | -90° | 0.707 | -90° | 3.89 | --- |  4.02 | 180° | 10 | -90° |
| 87612 | 1.22 | -90° | 0.013 | -178.95° | 0.0714 | -88.95° | 1 | 91.05° | 0.183 | -178.95° |

<center><img src="images/TINA_nPqYQ55L9F.png" width="400"/> <h4>Рисунок 2.15 - Графіки напруг VМ1, VМ2, VМ3 і VМ4</h4></center>

За отриманими графіками (рис. 2.16, 2.17) частота послідовного резонансу приблизно становить 8.67 кГц. 

<center><img src="images/TINA_6RTYKxxisq.png" width="400"/> <img src="images/TINA_IoDfH0Tl4C.png" width="400"/> <h4>Рисунок 2.16 - Графіки АЧХ та ФЧХ (Лінійна та Логарифмічна розгортки)</h4></center>

Приберемо зі схеми всі вольтметри, крім вольтметра VМ2. Через пункт меню «Аналіз – Об'єкт управління» зробимо резистор R1 керованим, тобто задамо йому чотири значення від 100 до 900 Ом з кроком 200 Ом. 

<center><img src="images/TINA_Xd6IKQJ2X2.png" width="400"/> <h4>Рисунок 2.17 - Графіки АЧХ та ФЧХ (При різних втратах в RLC ланцюзі)</h4></center>

Дивлячись на (рис. 2.17), можна зробити висновок, що при більшому опорі $R1$ струм в колі менше, а значить, $R1$ демпфує (гасить) коливання, заважаючи $L1$ та $C1$ ефективно обмінюватися енергією. Це робить резонансний пік нижчим і ширшим.

Підключимо до точок А і B схему з двома ємностями (рис. 2.17). Встановимо амплітуду сигналу генератора напруги 1 В. Отримаємо граіки АЧХ і ФЧХ двополюсника в діапазоні частот від 2 до 20 кГц (рис. 2.18).

<center><img src="images/TINA_GTo2Ec9PAi.png" width="300"/> <h4>Рисунок 2.17 - Схема з двома ємностями</h4></center>

<center><img src="images/TINA_wcF8Lg6euD.png" width="400"/> <h4>Рисунок 2.18 - Графіки АЧХ та ФЧХ (від 2 до 20 кГц)</h4></center>

В діапазоні частот від 2 до 20 кГц (з кроком через 2 кГц) виміряємо напруги і фази у вузлах А ($U_{VM1}$) і B ($U_{VM2}$). Результати запишемо в таблицю 2.2.

Таблиця вимірів та розрахунків 2.2
| Частота, Гц | $U_{VM1}$, В | $\varphi_{VM1}$ | $U_{VM2}$, В | $\varphi_{VM2}$ |
|:-----|:-----|:-----|:-----|:-----|
| 2000 | 1 | -90° | 0.06 | -4.87° |
| 4000 | 1 | -90° | 0.13 | -10.63° |
| 6000 | 1 | -90° | 0.337 | -19.73° |
| 8000 | 1 | -90° | 0.758 | -49.33° |
| 10000 | 1 | -90° | 0.430 | -154.48° |
| 12000 | 1 | -90° | 0.035 | -177.98° |
| 14000 | 1 | -90° | 0.102 | -5.9° |
| 16000 | 1 | -90° | 0.186 | -10.74° |
| 18000 | 1 | -90° | 0.25 | -14.43° |
| 20000 | 1 | -90° | 0.3 | -17.55° |

Для своєї схеми з триелементним двополюсником на частоті, f = 0,5*N + n + 2 кГц (N – номер підгрупи, а n – номер бригади), виміряємо напругу і фази в точках схеми А, B і D. Результати запишемо в таблицю 2.3.

$$
f = 0.5 \cdot N + n + 2000 =
$$

$$
f = 0.5 \cdot N + n + 2000 = 0.5 \cdot 1 + 2 + 2000 = 2002.5
$$

Таблиця вимірів та розрахунків 2.3
| Частота, Гц | $U_{A}$, В | $\varphi_{A}$ | $U_{B}$, В | $\varphi_{B}$ | $U_{D}$, В | $\varphi_{D}$ |
|:-----|:-----|:-----|:-----|:-----|:-----|:-----|
| 2002.5 | 1 | -90° | 0.085 | -4.88° | 1.05 | -90.25° |

Тепер розрахуємо напруги і фази в точках схеми А, B і D. Результати запишемо в таблицю 2.4.

Таблиця вимірів та розрахунків 2.4
| Частота, Гц | $U_{A}$, В | $\varphi_{A}$ | $U_{B}$, В | $\varphi_{B}$ | $U_{D}$, В | $\varphi_{D}$ |
|:-----|:-----|:-----|:-----|:-----|:-----|:-----|
| 2002.5 | 1 | -90° |  | ° |  | ° |

Запишемо рівняння для схеми в копмлесній формі:
$$
i_mR + i_m j\omega L + i_m \left(\frac{-j}{\omega C_1 + C_2}\right) = \dot{E}_m
$$

Запишемо рівняння для схеми в копмлесній формі:
$$
I_m \left(R + j\omega L + \frac{1}{j \omega (C_1 + C_2)}\right) = \dot{E}_m
$$

In [76]:
import sympy as smp
import numpy as nmp
import math
from sympy import lambdify

E_s = 1;
R_s = 100;
L_s = 0.01;
C1_s = 3.3 * 10 **-8;
C2_s = 3.3 * 10 **-8;
f_s = 2002.5;
j = smp.I

# Оголошуємо символьні змінні для формули
I, E, R, L, C1, C2, W, f, = smp.symbols('I E R L C1 C2 W f')

#eq = [I*((1/(1/((j*W*L)+(1/(j*W*C1)))+(1/(1/(j*W*C2)))))+R)-E]

eq = [E / (R + (1/(1/((j*W*L)+(1/(j*W*C1))+1/(1/(j*W*C2))))))-I]

sol = smp.solve(eq, [I])
sol[I]


V_b = sol[I] * R
V_d = E - (sol[I]*(j*W*L))

V_b_f = lambdify([E, R, L, C1, C2, W], V_b, 'numpy')
V_d_f = lambdify([E, R, L, C1, C2, W], V_d, 'numpy')

print(nmp.abs(V_b_f(E_s, R_s, L_s, C1_s, C2_s, 2*nmp.pi*f_s)))
print(nmp.abs(V_d_f(E_s, R_s, L_s, C1_s, C2_s, 2*nmp.pi*f_s)))

print(nmp.angle(V_b_f(E_s, R_s, L_s, C1_s, C2_s, 2*nmp.pi*f_s),deg=True))
print(nmp.angle(V_d_f(E_s, R_s, L_s, C1_s, C2_s, 2*nmp.pi*f_s),deg=True))


0.04376757934148486
1.055018695239895
87.49150110862159
-0.13089416987344982


In [77]:
sol[I]

-I*C1*E*W/(C1*C2*W**2 + C1*L*W**2 - I*C1*R*W - 1)

In [78]:
import numpy as np
from cmath import rect

E_s = 1
R_s = 100
L_s = 0.01
C1_s = C2_s = 3.3 * 10 ** -8
f_s = 2002.5
omega = 2 * np.pi * f_s

# Опори
Z_L = 1j * omega * L_s
Z_C1 = 1 / (1j * omega * C1_s)
Z_C2 = 1 / (1j * omega * C2_s)
Z_LC1 = Z_L + Z_C1
Z_eq = 1 / (1 / Z_LC1 + 1 / Z_C2)
Z_total = R_s + Z_eq

# Струм
I = E_s / Z_total

# Напруги
V_A = E_s
V_B = I * R_s
V_L = I * Z_L
V_D = V_A - V_L  # Напруга в точці D (після L1)

# Модулі та фази
print("Модуль V_B:", abs(V_B))
print("Модуль V_D:", abs(V_D))
print("Фаза V_B (градуси):", np.angle(V_B, deg=True))
print("Фаза V_D (градуси):", np.angle(V_D, deg=True))

Модуль V_B: 0.08502144059651957
Модуль V_D: 1.106624678733261
Фаза V_B (градуси): 85.1227422115753
Фаза V_D (градуси): -0.4709084489431899


#### Висновок
В результаті виконання лабораторної роботи були досліджені амплітудні та фазові співвідношення, а також резонансні явища в RLC-колах змінного струму. Розрахунки для послідовного коливального контуру показали, що теоретична резонансна частота становить 8.76 кГц, що добре узгоджується з експериментально отриманим піком на амплітудно-частотній характеристиці (АЧХ) на рівні ~8.67 кГц. Вимірювання напруг на реактивних елементах у режимі резонансу продемонстрували значне їх перевищення над напругою джерела живлення, що підтверджує явище резонансу напруг.

Аналіз впливу активного опору R1 (у діапазоні від 100 до 900 Ом) на характеристики контуру показав, що збільшення опору призводить до демпфування коливань, що виражається у зниженні та розширенні резонансного піку на АЧХ. Це підтверджує теоретичні положення про добротність коливального контуру. Теоретичні розрахунки вузлових напруг та фаз, виконані за допомогою методу комплексних амплітуд, показали відповідність результатам, отриманим у середовищі моделювання TINA.

Результати роботи демонструють точність теоретичних розрахунків та їхню практичну застосовність для аналізу та проектування електричних кіл змінного струму. Для підвищення точності в майбутніх дослідженнях рекомендується провести натурні експериментальні вимірювання для врахування неідеальності реальних компонентів.